src/custom/connectors/index.ts

```diff
@@ -1,16 +1,110 @@
-import { ChainId } from '@uniswap/sdk'
+import { Web3Provider } from '@ethersproject/providers'
 import { InjectedConnector } from '@web3-react/injected-connector'
+import { WalletConnectConnector } from '@web3-react/walletconnect-connector'
+import { WalletLinkConnector } from '@web3-react/walletlink-connector'
+import { PortisConnector } from '@web3-react/portis-connector'
 
-export {
-  NETWORK_CHAIN_ID,
-  fortmatic,
-  portis,
-  network,
-  walletconnect,
-  walletlink,
-  getNetworkLibrary
-} from '@src/connectors'
-
-export const injected = new InjectedConnector({
-  supportedChainIds: [ChainId.MAINNET, ChainId.ROPSTEN, ChainId.RINKEBY, ChainId.GÖRLI, ChainId.KOVAN, ChainId.XDAI]
+import { FortmaticConnector } from 'connectors/Fortmatic'
+import { NetworkConnector } from 'connectors/NetworkConnector'
+
+const SUPPORTED_CHAIN_IDS = process.env.REACT_APP_SUPPORTED_CHAIN_IDS
+const DEFAULT_CHAIN_ID = Number(process.env.REACT_APP_CHAIN_ID || '1')
+
+const FORMATIC_KEY = process.env.REACT_APP_FORTMATIC_KEY
+const PORTIS_ID = process.env.REACT_APP_PORTIS_ID
+
+export const NETWORK_CHAIN_ID: number = parseInt(process.env.REACT_APP_CHAIN_ID ?? '1')
+
+type RpcNetworks = { [chainId: number]: string }
+
+function getRpcNetworks(): [RpcNetworks, number[]] {
+  // Make sure the mandatory envs are present
+  if (typeof SUPPORTED_CHAIN_IDS === 'undefined') {
+    throw new Error(`REACT_APP_NETWORK_URL must be a defined environment variable`)
+  }
+
+  // Get list of supported chains
+  const chainIds = SUPPORTED_CHAIN_IDS.split(',').map(chainId => Number(chainId.trim()))
+  if (chainIds.length === 0) {
+    throw new Error(`At least one network should be supported. REACT_APP_CHAIN_ID`)
+  }
+
+  // Make sure the default chain is in the list of supported chains
+  if (!chainIds.includes(DEFAULT_CHAIN_ID)) {
+    throw new Error(
+      `The default chain id (${DEFAULT_CHAIN_ID}) must be part of the list of supported networks: ${chainIds.join(
+        ', '
+      )}`
+    )
+  }
+
+  // Return rpc urls per network
+  const rpcNetworks = chainIds.reduce<RpcNetworks>((acc, chainId) => {
+    const url = process.env['REACT_APP_NETWORK_URL_' + chainId]
+
+    if (typeof url === 'undefined') {
```

Comment:
curious why you are checking for just undefined vs just truthy, e.g
if (!url) - are you expecting a `0`, or `false` or `null` url as valid?

Reply:
yes, potencially I could do validations even that it has the right format. But this is meant to be a quick check if it's present.

However, if we want to be purist, note that the message refers to an env not being defined. So if a user sets it to false the message is not correct. Anyways,  I'll just change it for a `if (url)` to do a basic check.